### DISCUSSION TITLE
Improve/fix some built in themes

### DISCUSSION THREAD

**Discussion by ChrHorn** [helix-editor/helix#10516] [General]
Some themes that come with helix need some polish.

acme: statusline, menu, maybe errors
<img width="640" alt="acme" src="https://user-images.githubusercontent.com/17070041/182567256-c8a07456-ddfb-4b4e-a5bb-f17d921cecc0.png">

base_16_transparent: menu bg same as standard bg (intentional, add border once it's possible)
<img width="640" alt="base16_transparent" src="https://user-images.githubusercontent.com/17070041/182567914-afefa999-475d-4fba-b0f4-cd006af76fa2.png">

nord_light: ui menu selected bg very similar to standard bg
<img width="640" alt="nord_light" src="https://user-images.githubusercontent.com/17070041/182568910-d861a1ab-c4b9-4cc4-8eea-4fb4568a4d4d.png">

rose_pine_dawn: ui menu selected bg same as standard bg
<img width="640" alt="rose_pine_dawn" src="https://user-images.githubusercontent.com/17070041/182569160-3f82d8f8-278b-417a-9b14-6edf5e189dc8.png">

<details>
<summary>FIXED</summary>

dark_plus: indent guide bg
<img width="640" alt="dark_plus" src="https://user-images.githubusercontent.com/17070041/182568231-a6f07f79-1ec0-4996-9ab2-71327feffe6f.png">

ayu_light: statusline fg, menu fg
<img width="640" alt="ayu_light" src="https://user-images.githubusercontent.com/17070041/182567605-67a10256-2805-4fb6-98d8-6b566e7b50f4.png">
</details>

**Comment by AlexanderBrevig** [comment]
I agree, and made a tool for helping out #3234. A lot of the issues you point out are not yet checked against, but they could :)

**Comment by GreasySlug** [comment]
About base16_transparent, it is intentionally the same as the background.
The reason for this is to maintain the uniformity of transparency.
I agree that it is hard to see, but it is not possible to change the transparency  #2751 or add a border of the pop-up background.

**Comment by ChrHorn** [comment]
Some of these might be intentional and fixes would be a bit opinionated. Pinging

@two-six for Acme, Nord Light
@enkodr for Ayu Light
@chunghha for Rose Pine Dawn

> About base16_transparent, it is intentionally the same as the background. The reason for this is to maintain the uniformity of transparency. I agree that it is hard to see, but it is not possible to change the transparency #2751 or add a border of the pop-up background.

Makes sense, added a note.

**Comment by xcdkz** [comment]
Thanks for the ping, I'll look into fixing acme.
The main point of the nord light is the low contrast between elements of the ui, so in my opinion it looks just like it should although I'll try experimenting a little with it.

**Comment by AlexanderBrevig** [comment]
> Thanks for the ping, I'll look into fixing acme. The main point of the nord light is the low contrast between elements of the ui, so in my opinion it looks just like it should although I'll try experimenting a little with it.

It is very hard to read the undefined `["warning","error","info","hint","diagnostic"]`.

![image](https://user-images.githubusercontent.com/721090/182810236-6fde545f-39b4-4406-afe7-37aea73389fa.png)

**Comment by ChrHorn** [comment]
> Thanks for the ping, I'll look into fixing acme. The main point of the nord light is the low contrast between elements of the ui, so in my opinion it looks just like it should although I'll try experimenting a little with it.

I think the main issue is that the code and the selected element blend into each other a bit too much. The contrast looks fine for the standard nord theme:
<img width="640" alt="nord" src="https://user-images.githubusercontent.com/17070041/182813200-cec91d6c-16f2-44d8-b668-272ade3e710e.png">
Looks also fine in the VSCode Nord Light theme:
![image](https://user-images.githubusercontent.com/17070041/182814988-37858dca-7560-4f25-bee7-c7a7d174ddc0.png)

**Comment by xcdkz** [comment]
Are the colors terminal-dependent? I have no problems in Konsole, can you tell me what terminal do you use?
![scrot helix](https://user-images.githubusercontent.com/92624829/182946355-c9c4906f-d5d7-45ad-a014-c7df69569b35.png)

**Comment by AlexanderBrevig** [comment]
It maybe/probably falls back to some base16 color set for the terminal? I use a gruvbox theme.

```
| alacritty | wezterm |
|   Foot    |  kitty  |
```

![image](https://user-images.githubusercontent.com/721090/182951062-d6164370-828a-4851-b039-a975afc92bb7.png)

**Comment by ChrHorn** [comment]
Acme uses a mix of hard coded colors and colors derived from the terminal theme. That's why it can look pretty bad when a dark terminal theme is used.

The issue for the Nord Light errors is that you used `diagnostic.error` instead of `error`. `error` will color the gutter and the overlaid text, `diagnostic.error`  will only color the text.

**Comment by yelfathi** [comment]
Using Acme theme on iterm2 (acme theme also), and I got this white background for errors don't know if it is related to this issue. Thanks!

EDIT: also the buffer line background.

<img width="1133" alt="image" src="https://user-images.githubusercontent.com/46417673/211568511-09a8d7f1-570b-42d3-82a2-5a5edf80622f.png">

**Comment by xcdkz** [comment]
@olbitset I have already fixed this in #5019, it just wasn't included in the latest release. You would need to compile helix from source or update `$HOME/.config/helix/runtime/themes/acme.toml` file to the [current version](https://github.com/helix-editor/helix/blob/master/runtime/themes/acme.toml).

**Comment by yelfathi** [comment]
Thanks @two-six I included as a personal theme. What about the TOP bufferline (as the selection one), is it expected to have a gray background? If so what is the parameter in the toml config to adapt it to my need?

**Comment by xcdkz** [comment]
> Thanks @two-six I included as a personal theme. What about the TOP bufferline (as the selection one), is it expected to have a gray background? If so what is the parameter in the toml config to adapt it to my need?

Do you mean foreground? I see background is light blue in your screenshot, not gray. 

Set `"ui.bufferline" = { fg = "indent", bg = "acme_bar_bg" }` to `"ui.bufferline" = { fg = "black", bg = "acme_bar_bg" }`.
 If you want you can put any color you want there instead of `black`, like this: `"ui.bufferline" = { fg = "#000000", bg = "acme_bar_bg" }`
 
 I've never used bufferline before, but as you mentioned current configuration doesn't really fit this theme really well. I will open a pull request to change it from gray to black.

**Comment by yelfathi** [comment]
Sorry I was talking about the bg. Thanks for the inputs I will adapt to my preferences!
